lib/qam.pm

```diff
@@ -74,16 +74,43 @@ sub is_patch_needed {
 # Function that will add all test repos
 sub add_test_repositories {
     my $counter = 0;
+
+    my $oldrepo = get_var('PATCH_TEST_REPO', '');
     my @repos = split(/,/, get_var('MAINT_TEST_REPO', ''));
+    # Be carefull. If you have defined both variables, the PATCH_TEST_REPO variable will always
+    # have precedence over MAINT_TEST_REPO. So if MAINT_TEST_REPO is required to be installed
+    # please be sure that the PATCH_TEST_REPO is empty.
+    @repos = split(',', $oldrepo) if ($oldrepo);
+
     for my $var (@repos) {
         zypper_call("--no-gpg-check ar -f $var 'TEST_$counter'");
         $counter++;
     }
-
     # refresh repositories, inf 106 is accepted because repositories with test
     # can be removed before test start
     zypper_call('ref', exitcode => [0, 106]);
 }
 
+# Function that will remove all test repos
+sub remove_test_repositories {
+
+    my $output   = script_output("zypper lr");
+    my @repoList = ();
+    for my $line (split(/\n/, $output)) {
+        my $repo = (split(/\|/, $line))[1];
+        if ($repo =~ /TEST_\d+/) {
+            $repo =~ s/^\s+|\s+$//g;
+            push @repoList, $repo;
+        }
+    }
+    zypper_call('rr ' . join(' ', @repoList)) if @repoList;
```

Comment:
This case won't fail:

`repos=($(zypper lr -e - | grep name=TEST | cut -d= -f2));  echo $repos; if [ ${#repos[@]} -ne 0 ]; then zypper rr $repos; fi`